Oracle.md

```diff
@@ -0,0 +1,196 @@
+        - [Version 0 `oracle_announcement`](#version-0-oracle_announcement)
+
+
+## Event descriptor
+
+An event descriptor provides information to clients about an event for which an oracle plans on releasing a signature over its outcome.
+The provided information should be sufficient for a client to create a set of CETs that will cover all possible outcomes of the event.
+
+Here we assume that an event outcome can be represented either as a set of strings or numbers.
+Four kinds of event outcomes are defined (note that a single event can be composed of several types).
+
+### Simple enumeration
+
+For events that have a narrow range of possible outcomes, the outcomes can simply be enumerated.
+
+#### Example: Weather tomorrow
+
+Tomorrow's weather can be represented as the set of strings `[sunny, cloudy, rainy]`.
+
+### Range
+
+When an event has numerical outcomes that cannot be easily enumerated, they can be represented as a series with:
+
+- start: the first possible outcome number
+- count: the number of possible outcomes
+- step: the increment
+- unit: the unit of the outcome value
+- precision: the precision of the outcome representing the base 10 exponent by which to multiply the signed number to obtain the actual outcome value.
+
+#### Example: tomorrow's temperature
+
+```
+start: -100
+count: 201
+step: 1
+unit: Celcius degrees
+precision: 0
+```
+
+### Large range
+
+When a range of a numerical outcomes is large or unbounded, the oracle can represent the outcome using numerical decomposition.
+
+The event descriptor should include:
+
+- base: a number representing the base in which the outcome value is decomposed and which will indicate the possible range of each digit that will be signed (e.g. 2 for binary, 10 for decimal, 16 for hex-decimal).
+- is-signed: a boolean value indicating whether the outcomes can be negative.
+- r-values: an array of R values, one for each of the digit that will be signed by the oracle, with the first value of the array being used for signing the leftmost digit. If the is-signed value is true, an additional R-value must be provided as the first element in this array, which will be used to sign the string "+" in case of an outcome with a positive value or zero value and the string "-" in case of an outcome with a negative one.
+- unit: the unit of the outcome value
+- precision: the precision of the outcome representing the base exponent by which to multiply the number represented by the composition of the digits to obtain the actual outcome value.
+
+In the case where the number of digits that make up the outcome value exceeds the number of r-values that the oracle committed to, the oracle should sign the maximum possible value that he can attest to. 
+
+When the number of digits that make up the outcome is lower than the number of r-values that the oracle committed to, the oracle should sign and publish a signature over the number zero with the extra R-values.
```

Comment:
This wording is a little ambiguous as someone signing 4 digits for the outcome 123 might interpret this as `1230` instead of `0123`.

Perhaps something like: "When the outcome is not the same order of magnitude as the maximum possible outcome, leading zeros must be signed."

Reply:
Done